src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        str += '';
+      } else {
+        str += this[i];
+      }
```

Comment:
no needs to add this else condition
```suggestion
        str += this[i];
      }
```